### PR TITLE
Jolokia ts+fetch

### PR DESCRIPTION
**Pull request by grgrzybek**

### CODE REVIEW

**Review by tadayosi — APPROVED** [review]

File: client/javascript-esm/packages/jolokia/src/jolokia.ts [hunk 1]
```diff
@@ -0,0 +1,1185 @@
...
+/**
+ * For POST requests it is recommended to have a trailing slash at the URL in order to avoid a redirect which then
+ * results in a GET request.
+ * @see https://bugs.eclipse.org/bugs/show_bug.cgi?id=331194#c1
+ * @param url Jolokia agent URL
+ * @returns URL location with "/" appended
+ */
+function ensureTrailingSlash(url: string | URL): string {
+  // Squeeze any URL to a single one, optionally adding one
+  return (typeof url === "string" ? url : url.href).replace(/\/*$/, "/")
```
[the remark this answers is not in the document]
Comment (grgrzybek) [reply client/javascript-esm/packages/jolokia/src/jolokia.ts:907]: oh my...

[the remark this answers is not in the document]
Comment (tadayosi) [reply client/javascript-esm/packages/jolokia/src/jolokia.ts:907]: For the first time in Jolokia I see the CodeQL review is working :-)

[the remark this answers is not in the document]
Comment (grgrzybek) [reply client/javascript-esm/packages/jolokia/src/jolokia.ts]: fixed by dropping regexps

[the remark this answers is not in the document]
Comment (grgrzybek) [reply client/javascript-esm/packages/jolokia/src/jolokia.ts]: I didn't turn on anything ;)

**Pull request merged**